### PR TITLE
Input

### PR DESCRIPTION
**Pull request by Lanc3**
comment me bitch

### CODE REVIEW

File: spikes/input spike/InputHandler.cpp [hunk 1]
```diff
@@ -0,0 +1,269 @@
...
+using namespace std;
+//this class handles the inputs of the controller to be used outside this class cleanly
+
+//if (inputHandler.getReleasedButton() == InputHandler::BUTTON::A)  <-- use the method to get the key pressed in your class outside this one
+//{}
+InputHandler::InputHandler()
+{
+	wasButtonAPressed = false;
+	wasButtonBPressed = false;
+	wasButtonXPressed = false;
```
Comment (gdxn96) [inline spikes/input spike/InputHandler.cpp:12]: c++ use your initializer list for simple initialization

File: spikes/input spike/InputHandler.cpp [hunk 2]
```diff
@@ -0,0 +1,269 @@
...
+	wasButtonYPressed = false;
+	wasButtonLeftBumpPressed = false;
+	wasButtonRightBumpPressed = false;
+	wasButtonStartPressed = false;
+	wasButtonSelectPressed = false;
+	lastDpadState = DPadState::CENTER;
+	debug = true;
+}
+void InputHandler::HandleInput()
+{
```
Comment (gdxn96) [inline spikes/input spike/InputHandler.cpp:22]: at least one comment above every method

File: spikes/input spike/InputHandler.cpp [hunk 3]
```diff
@@ -0,0 +1,269 @@
+#include "stdafx.h"
```
Comment (gdxn96) [inline spikes/input spike/InputHandler.cpp:1]: //Class name
//Short Class Description
//
//At head of every cpp

File: spikes/input spike/InputHandler.cpp [hunk 4]
```diff
@@ -0,0 +1,269 @@
...
+	}
+	else if(isButtonXPressed && !wasButtonXPressed)
+	{	
+		if(debug)
+		{
+			printf("button X : \n");
+		}
+		pressedButton = BUTTON::X;
+	}
+	else if(isButtonYPressed && !wasButtonYPressed)
```
Comment (gdxn96) [inline spikes/input spike/InputHandler.cpp:62]: group checks into methods, too messy in one function, also two buttons pressed at the same time may not work with this code

File: spikes/input spike/Vector2D.cpp [hunk 5]
```diff
@@ -0,0 +1,157 @@
...
+Vector2D Vector2D::Perpendicular(const Vector2D &v)
+{
+    return Vector2D(v.y, -v.x);
+}
+Vector2D Vector2D::DirectionVectorBetweenTwoPoints(const Vector2D &v1, const Vector2D & v2)
+{
+	Vector2D vec = Vector2D(v2.x-v1.x,v2.y-v1.y);
+	return vec.Normalize();
+}
+bool Vector2D::Intersect(const Vector2D &aa, const Vector2D &ab, const Vector2D &ba, const Vector2D &bb)
```
Comment (gdxn96) [inline spikes/input spike/Vector2D.cpp:133]: again comment your functions

File: spikes/input spike/InputHandler.cpp (hunk 4 above)
Comment (Lanc3) [inline spikes/input spike/InputHandler.cpp:62]: two buttons are not meant to be pressed with this function. "pressedOnce" its the base function that is build on. I.E the next step is to store a list of buttons pressed in sequence. list clears after a set time. this allows for combos. Also will separate parts in to other methods for clarity.

File: spikes/input spike/Vector2D.cpp (hunk 5 above)
Comment (Lanc3) [inline spikes/input spike/Vector2D.cpp:133]: comments in header file

File: spikes/input spike/InputHandler.cpp (hunk 1 above)
Comment (Lanc3) [inline spikes/input spike/InputHandler.cpp:12]: can you give me an example of what you mean

Comment (gdxn96) [inline spikes/input spike/InputHandler.cpp:12]: yes, 
InputHandler::InputHandler() : wasButtonAPressed(false), wasButtonBPressed(false)
{

do this with all constructor values you can, only move it down if its complex.

**Comment by gdxn96** [comment]
This is one of the few classes that need to be perfect before the merge, so I'm closing this pull request temporarily and treating it like a first iteration, when we have an idea what we want with controls this branch can then be edited and merged

**Pull request merged**